src/utils/image_augmentor.py

```diff
@@ -14,28 +12,69 @@ def __init__(self, config):
     self.load_annotations()
     self.init_file_list()
 
+  # Helper function to correct float rounding errors in LabelStudio
+  # ie. emission of values with percentages > 100 or < 0
+  def fp_adjust(self, dim):
+    if dim < 0:
+      return 0.0
+    elif dim > 100:
+      return 100.0
+    else:
+      return dim
+
   # Augment an image to the expected configuration, saving the new versions to an configured output path
   def process(self, path):
     with Image.open(path) as img:
       img, rotate_type = self.aug_rotation(img)
       img, satur_type = self.aug_saturation(img)
-
       output_path = self.build_output_path(path, [rotate_type, satur_type])
       # skip saving file and adding annotation if one exists with the same name
       if str(output_path.resolve()) in self.path_to_anno:
         return output_path
       # construct path to write to, then save the file
       output_path.parent.mkdir(parents=True, exist_ok=True)
       img.save(output_path, "JPEG", optimize=True, quality=80)
-
-      self.add_aug_annotation(path, output_path)
+      self.add_aug_annotation(path, output_path, rotate_type)
       self.add_to_file_list(output_path)
       return output_path
 
-  def add_aug_annotation(self, orig_path, output_path):
+  # Generates augmented annotation by recomputing dimensions and labels after rotation.
+  def add_aug_annotation(self, orig_path, output_path, rotate_type):
     orig_anno = self.path_to_anno[str(orig_path.resolve())]
     aug_anno = copy.deepcopy(orig_anno)
+    # Populate augmented annotation
     aug_anno['image'] = str(output_path)
+    aug_anno['rotation_type'] = rotate_type
+    for label in aug_anno['label']:
```

Comment:
I tried running this on longleaf and got the following error for file `/work/users/b/b/bbpennel/color_bars/norms/opt/data/color_bars/gilmer/00276_op0217_0001_e.jpg` (annotation id 471):
```
Traceback (most recent call last):
  File "/work/users/b/b/bbpennel/git/ml-repo-preingest-processing/augment_images.py", line 55, in <module>
    augmentor.process(path)
  File "/work/users/b/b/bbpennel/git/ml-repo-preingest-processing/src/utils/image_augmentor.py", line 37, in process
    self.add_aug_annotation(path, output_path, rotate_type)
  File "/work/users/b/b/bbpennel/git/ml-repo-preingest-processing/src/utils/image_augmentor.py", line 48, in add_aug_annotation
    for label in aug_anno['label']:
KeyError: 'label'
```
it may be that we need an early return in add_aug_annotation if there are no labels in an annotation.

Reply:
Good catch, fixed.